### PR TITLE
REF: formats.format functions that should be DTA methods

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -735,14 +737,35 @@ def astype(self, dtype, copy: bool = True):
...
         self, *, na_rep: str | float = "NaT", date_format=None, **kwargs
     ) -> npt.NDArray[np.object_]:
-        from pandas.io.formats.format import get_format_datetime64_from_values
-
-        fmt = get_format_datetime64_from_values(self, date_format)
+        if date_format is None:
+            ido = self._is_dates_only
+            if ido:
+                # Only dates and no timezone: provide a default format
+                date_format = "%Y-%m-%d"
```
Comment (mroeschke) [inline pandas/core/arrays/datetimes.py:744]:
```suggestion
        if date_format is None and self._is_dates_only:
            # Only dates and no timezone: provide a default format
            date_format = "%Y-%m-%d"
```

Nit

Comment (jbrockmendel) [reply pandas/core/arrays/datetimes.py:744]: good idea, will update

Comment (jbrockmendel) [reply pandas/core/arrays/datetimes.py]: #55405 implements this same method in the mixin class TimelikeOps.  That is the better long-term location.  Will update whichever PR goes in second.

**Comment by jbrockmendel** [comment]
updated+green. this and #55405 are blockers for several in-progress branches.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**